straxen/plugins/pulse_processing.py

```diff
@@ -62,13 +62,8 @@
              'Specify as a tuple of length n_tpc_pmts, or a number.'),
 
     strax.Option(
-        'n_tpc_pmts',
-        default=straxen.n_tpc_pmts,
+        'n_tpc_pmts', type=int,
```

Comment:
For some reason I'm not able to acces this option. I'm not able to get it to the ```self.config``` by e.g. passing on these options: ```{'daq_input_dir': '/live_data/xenonnt/007079', 'daq_compressor': 'lz4', 'run_start_time': 1583745091.405, 'record_length': 500, 'n_readout_threads': 30, 'n_tpc_pmts': 493, 'channel_map': <frozendict {'tpc': (0, 493), 'lowgain': (500, 752), 'aqmon': (799, 807), 'tpc_blank': (999, 999), 'mv': (1000, 1083), 'mv_blank': (1999, 1999), 'diagnostic': (3000, 4000)}>}```

Reply:
Interesting: since no default is given, something must be getting to this option to allow the Travis build to succeed (at least in the XENON1T condition). But you must be passing the config correctly, otherwise the daqreader would have complained about missing daq_input_dir... Not sure what is going on.

Reply:
Could it be due to strax? If I add a default everyting works. I'm not sure what happens if no default is given here:
https://github.com/AxFoundation/strax/blob/master/strax/config.py

Reply:
I see you're only adding the `xnt_common_config` options inside `st_make`, rather than in `new_context`, maybe this is an issue. Otherwise, could you post the exact error you're getting?

I'm happy to dig for bugs in strax's config handling. It's just curious that `daq_input_dir` also doesn't have a default, but seems to be passed fine, and the fake DAQ has no trouble finding `n_tpc_pmts`.

Reply:
> I see you're only adding the `xnt_common_config` options inside `st_make`, rather than in `new_context`, maybe this is an issue. Otherwise, could you post the exact error you're getting?
> I'm happy to dig for bugs in strax's config handling. It's just curious that `daq_input_dir` also doesn't have a default, but seems to be passed fine, and the fake DAQ has no trouble finding `n_tpc_pmts`.

Jelle you are a genius! It does work now. I must admit I'm quite surprised that as you say the ```daq_input_dir``` can be passed as an argument this way whereas the ```n_tpc_pmts``` apparantly cannot?

Reply:
Great! I think n_tpc_pmts is probably already used in some infer_dtype, so crash will complain about that at an earlier stage. Do you want to push the bootstrax fix to this PR? And are you OK to merge after that?

Reply:
Sure, I'm merging after Travis approves